iofog-agent-daemon/src/test/java/org/eclipse/iofog/message_bus/CommandLineHandlerTest.java

```diff
@@ -0,0 +1,171 @@
+        ClientSession.class, ClientProducer.class})
+public class CommandLineHandlerTest {
+    private String MODULE_NAME;
+    private CommandLineHandler commandLineHandler;
+    private ClientMessage message;
+    private ClientSession clientSession;
+    private ClientProducer clientProducer;
+
+    @Before
+    public void setUp() throws Exception {
+        MODULE_NAME = "CommandLineHandler";
+        commandLineHandler = spy(new CommandLineHandler());
+        message = mock(ClientMessage.class);
+        clientSession = mock(ClientSession.class);
+        clientProducer = mock(ClientProducer.class);
+        mockStatic(LoggingService.class);
+        mockStatic(CommandLineParser.class);
+        mockStatic(MessageBusServer.class);
+        when(CommandLineParser.parse(anyString())).thenReturn("success");
+        when(MessageBusServer.getSession()).thenReturn(clientSession);
+        when(MessageBusServer.getCommandlineProducer()).thenReturn(clientProducer);
+        when(clientSession.createMessage(false)).thenReturn(message);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        MODULE_NAME = null;
+    }
+
+    /**
+     * Test onMessage when message is null
+     */
+    @Test
+    public void testOnMessageWhenMessageIsNull() {
+        commandLineHandler.onMessage(null);
+        verifyStatic(LoggingService.class, Mockito.atLeastOnce());
+        LoggingService.logInfo(MODULE_NAME, "starting onMessage : listener for command-line communications");
+        verifyStatic(LoggingService.class, Mockito.never());
+        LoggingService.logInfo(MODULE_NAME, "starting onMessage : acknowledge");
+        verifyStatic(LoggingService.class, Mockito.atLeastOnce());
+        LoggingService.logInfo(MODULE_NAME, "Started onMessage : listener for command-line communications");
+    }
+
+    /**
+     * Test onMessage when message is empty
+     */
+    @Test
+    public void testOnMessageWhenMessageIsEmpty() {
+        commandLineHandler.onMessage(message);
+        verifyStatic(LoggingService.class, Mockito.atLeastOnce());
+        LoggingService.logInfo(MODULE_NAME, "starting onMessage : listener for command-line communications");
+        verifyStatic(LoggingService.class, Mockito.atLeastOnce());
+        LoggingService.logInfo(MODULE_NAME, "starting onMessage : acknowledge");
+        verifyStatic(LoggingService.class, Mockito.never());
+        LoggingService.logInfo(MODULE_NAME, "Parsing command");
+        verifyStatic(LoggingService.class, Mockito.atLeastOnce());
+        LoggingService.logInfo(MODULE_NAME, "Started onMessage : listener for command-line communications");
+    }
+
+    /**
+     * Test onMessage when message is valid
+     */
+    @Test
+    public void testOnMessageWhenMessageIsValid() {
+        when(message.getStringProperty(anyString())).thenReturn("command");
+        commandLineHandler.onMessage(message);
```

Comment:
instead of checking if logInfo, logError, ... is being called, you should expect for example:
- `message.acknowledge` is being called
- `message.getStringProperty` gets called with `command`
- CommandLineParser.parse with `command`
- ...
Other tests should follow the same pattern. You can check log messages when method under test does not call any other methods.

Reply:
Updated code as per the feedback.